src/transformers/visitors/orchestrationInternalFunctionCallVisitor.ts

```diff
@@ -213,6 +214,14 @@ const internalCallVisitor = {
                 file.nodes.forEach(childNode => {
                  if(childNode.nodeType === 'FunctionDefinition') {
                    childNode.parameters.modifiedStateVariables = joinWithoutDupes(childNode.parameters.modifiedStateVariables, state.newParametersList);
+                   const modifiedNodes = childNode.parameters.modifiedStateVariables.map(node => node.name);
+                   const decNode = childNode.body.preStatements.filter(node => node.nodeType === 'VariableDeclarationStatement');
+                   !JSON.stringify(decNode).includes(JSON.stringify(state.decNode)) ?
+                   state.decNode?.forEach((decNode, decIndex) => {
+                      // if the function doesn't modifies the returned variable don't include it
+                      if(state.decNode && !(modifiedNodes.includes(decNode.declarations[0].name))) 
+                      childNode.body.preStatements.splice(decIndex+1, 0, decNode); 
+                   }): '';
```

Comment:
The name decNode is used twice here in the for loop and also declared on line 218, could we use a different name for one?

Reply:
changed the name